File: src/main/java/com/thesilentnights/easylogin/commands/admin/EmailTest.java
package com.thesilentnights.easylogin.commands.admin;

import com.mojang.brigadier.arguments.StringArgumentType;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import com.mojang.brigadier.context.CommandContext;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.commands.Commands;

public class EmailTest extends AdminCommands {


    @Override
    public LiteralArgumentBuilder<CommandSourceStack> getCommand() {
        return super.MAIN_NODE.then(
                Commands.literal("email")
                        .then(
                                Commands.literal("test")
                                        .then(
                                                Commands.argument("targetEmail", StringArgumentType.greedyString())
                                                        .executes((CommandContext<CommandSourceStack> context) -> 1)
                                        )
                        )
        );
    }
}

File: src/main/kotlin/com/thesilentnights/easylogin/commands/admin/EmailTest.kt
package com.thesilentnights.easylogin.commands.admin

import com.mojang.brigadier.arguments.StringArgumentType
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.mojang.brigadier.context.CommandContext
import net.minecraft.commands.CommandSourceStack

class EmailTest : AdminCommands {
    override fun getCommand(mainNode: LiteralArgumentBuilder<CommandSourceStack>): LiteralArgumentBuilder<CommandSourceStack> {
        return mainNode.then(
            net.minecraft.commands.Commands.literal("email")
                .then(
                    net.minecraft.commands.Commands.literal("test")
                        .then(
                            net.minecraft.commands.Commands.argument<kotlin.String?>(
                                "targetEmail",
                                StringArgumentType.greedyString()
                            )
                                .executes { context: CommandContext<CommandSourceStack?>? ->
                                    //TODO: Send email
                                    1
                                }
                        )
                )
        )
    }
}
